### PR TITLE
feat(es/lint): support ignoreReadBeforeAssign for prefer-const rule

### PR DESCRIPTION
**Pull request by ArturAralin**
**Description:**

support ignoreReadBeforeAssign for prefer-const rule

**BREAKING CHANGE:**
No

### CODE REVIEW

File: crates/swc_ecma_lints/src/rules/prefer_const.rs [hunk 1]
```diff
@@ -11,15 +12,21 @@ use crate::{
 // todo: implement option destructuring: all | any
 // https://eslint.org/docs/rules/prefer-const#destructuring
 
-// todo: implement option ignoreReadBeforeAssign
-// https://eslint.org/docs/rules/prefer-const#ignorereadbeforeassign
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PreferConstConfig {
+    ignore_read_before_assign: Option<bool>,
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/prefer_const.rs:18]: I think we need `#[serde(default)]`

Comment (kdy1) [reply crates/swc_ecma_lints/src/rules/prefer_const.rs:18]: And it can be `bool` instead of `Option<bool>`.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

File: crates/swc_ecma_lints/src/rules/prefer_const.rs (hunk 1 above)
Comment (ArturAralin) [reply crates/swc_ecma_lints/src/rules/prefer_const.rs]: Sorry for late fix. I was busy

**Comment by kdy1** [comment]
Sorry for the delay.
We need some more discussion about linter

**Comment by ArturAralin** [comment]
> We need some more discussion about linter

With me?

**Comment by kdy1** [comment]
Ah, my bad. 
I meant the core team

**Comment by ArturAralin** [comment]
@kdy1 have any solution?

**Review by kdy1 — APPROVED** [review]
Thank you and sorry for the dealy

---



swc-bump:
 - swc_ecma_lints

**Pull request merged**